pkg/util/docker.go

```diff
@@ -3,22 +3,25 @@ package util
 import (
 	"encoding/json"
 	"io"
+	"os"
 
 	"github.com/docker/docker/pkg/jsonmessage"
 )
 
-// PipeDockerOutput pipes a reader that spits out jsonmessage structs into a
-// writer, usually stdout. It returns normally when the reader is exhausted, or
-// in error if one occurs.
-func PipeDockerOutput(r io.ReadCloser, w io.Writer) error {
+// PipeDockerOutput pipes a reader that spits out jsonmessage structs into one
+// or many writer, usually stdout. It returns normally when the reader is exhausted,
+// or in error if one occurs.
+func PipeDockerOutput(r io.ReadCloser, writers ...io.Writer) error {
 	var msg jsonmessage.JSONMessage
 Loop:
 	for dec := json.NewDecoder(r); ; {
 		switch err := dec.Decode(&msg); err {
 		case nil:
-			_ = msg.Display(w, true)
-			if msg.Error != nil {
-				return msg.Error
+			for _, w := range writers {
+				_ = msg.Display(w, w == os.Stdout || w == os.Stderr)
```

Comment:
Ah, I see that you need some fine-grained control here.

Reply:
Same as previous comment!